File: serde-generate/runtime/dart/bincode/bincode_serializer.dart
// Copyright (c) Facebook, Inc. and its affiliates
// SPDX-License-Identifier: MIT OR Apache-2.0

part of 'bincode.dart';

class BincodeSerializer extends BinarySerializer {
  BincodeSerializer()
      : super(
          containerDepthBudget: maxContainerDepth,
        );

  @override
  void serializeLength(int len) {
    serializeUint64(Uint64(BigInt.from(len)));
  }

  @override
  void serializeVariantIndex(int index) {
    serializeUint32(index);
  }

  @override
  void sortMapEntries(List<int> offsets) {
    // Not required by the format.
  }
}

File: serde-generate/runtime/kotlin/com/novi/bincode/BincodeSerializer.kt
// Copyright (c) Zefchain Labs, Inc.
// SPDX-License-Identifier: MIT OR Apache-2.0

package com.novi.bincode

import com.novi.serde.BinarySerializer
import com.novi.serde.SerializationError

class BincodeSerializer : BinarySerializer(Long.MAX_VALUE) {
    @Throws(SerializationError::class)
    override fun serialize_f32(value: Float) {
        serialize_i32(value.toRawBits())
    }

    @Throws(SerializationError::class)
    override fun serialize_f64(value: Double) {
        serialize_i64(value.toRawBits())
    }

    @Throws(SerializationError::class)
    override fun serialize_len(value: Long) {
        serialize_u64(value.toULong())
    }

    @Throws(SerializationError::class)
    override fun serialize_variant_index(value: Int) {
        serialize_u32(value.toUInt())
    }

    override fun sort_map_entries(offsets: IntArray) {
        // Not required by the format.
    }
}
